extensions/narayana-jta/deployment/src/main/java/io/quarkus/narayana/jta/deployment/NarayanaJtaProcessor.java

```diff
@@ -135,6 +138,15 @@ public void build(NarayanaJtaRecorder recorder,
         recorder.setConfig(transactions);
     }
 
+    @BuildStep
+    @Record(RUNTIME_INIT)
+    @Consume(NarayanaInitBuildItem.class)
+    public void startRecoveryService(NarayanaJtaRecorder recorder,
+            List<JdbcDataSourceBuildItem> jdbcDataSourceBuildItems, TransactionManagerConfiguration transactions) {
```

Comment:
I change to verify the `jdbcStoreDataSource` and start recovery service after all the datasouces are ready by adding a new `BuildStep`. This should resollve the cycle dependency of `agroal`.

Reply:
Very good.